### PR TITLE
Enable etcd experimental corruption checks

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Enable etcd experimental corruption checks.

**Release note**:
```release-note
Enable etcd corruption checks
```

### CODE REVIEW

File: api/pkg/resources/etcd/statefulset.go [hunk 1]
```diff
@@ -352,6 +352,8 @@ exec /usr/local/bin/etcd \
     --client-cert-auth \
     --cert-file /etc/etcd/pki/tls/etcd-tls.crt \
     --key-file /etc/etcd/pki/tls/etcd-tls.key \
-    --auto-compaction-retention=8
+    --auto-compaction-retention=8 \
+    --experimental-initial-corrupt-check=true \
```
Comment (alvaroaleman) [inline api/pkg/resources/etcd/statefulset.go:355]: Not sure How good I feel about flags that are marked experimental

Comment (mrIncompetent) [reply api/pkg/resources/etcd/statefulset.go:355]: For me the same, i would like to have this enabled on cloud for some time to see if we encounter any negative impact.

Comment (mrIncompetent) [reply api/pkg/resources/etcd/statefulset.go:355]: How about a feature flag?

Comment (alvaroaleman) [reply api/pkg/resources/etcd/statefulset.go:355]: SGTM

File: api/pkg/resources/etcd/statefulset_test.go [hunk 2]
```diff
@@ -40,110 +51,8 @@ func TestGetEtcdCommand(t *testing.T) {
...
-				t.Errorf("expected \n%s \n\nGot:\n\n%s", test.expected, cmd)
-			}
+
+			testhelper.CompareOutput(t, fmt.Sprintf("etcd-command-%s", test.name), cmd, *update)
 		})
 	}
 }
-
-var (
-	noMigration = `export MASTER_ENDPOINT="https://etcd-0.etcd.cluster-lg69pmx8wf.svc.cluster.local:2379"
```
Comment (mrIncompetent) [inline api/pkg/resources/etcd/statefulset_test.go:51]: Moved them to files for readability

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

**Comment by mrIncompetent** [comment]
@kron4eg Seems I changed something in the monitoring controller - would you mind doing a sanity check?

**Pull request merged**